AppServices/CustomerAppService.cs

```diff
@@ -8,30 +10,39 @@ namespace AppServices
     public class CustomerAppService : ICustomerAppService
     {
         private readonly ICustomerServices _customerServices;
+        private readonly IMapper _mapper;
 
-        public CustomerAppService(ICustomerServices customerServices)
+        public CustomerAppService(ICustomerServices customerServices, IMapper mapper)
         {
             _customerServices = customerServices;
+            _mapper = mapper;
         }
 
-        public IEnumerable<Customer> GetAll(Func<Customer, bool> predicate = null)
+        public IEnumerable<ReadCustomerDto> GetAll(Func<Customer, bool> predicate = null)
         {
-            return _customerServices.GetAll(predicate);
+            var customers = _customerServices.GetAll(predicate);
+
+            return customers is not null
+                ? _mapper.Map<IEnumerable<ReadCustomerDto>>(customers)
+                : null;
```

Comment:
Não faz diferença aqui
```suggestion
            return _mapper.Map<IEnumerable<ReadCustomerDto>>(customers);
```